source/task_executor.py

```diff
@@ -0,0 +1,171 @@
+import datetime
+from multiprocessing import Pool
+from typing import Generator, List, NamedTuple, Tuple
+
+import numpy as np
+from ont_fast5_api.fast5_interface import get_fast5_file, check_file_type
+from ont_fast5_api.fast5_read import Fast5Read
+
+from interface import BasecallerInterface, TaskExecutorInterface, FilePath, OutputData
+
+
+class CallInputData(NamedTuple):
+    """
+    Class is a container of input data for _call_read() function.
+    Design is motivated by the fact that ONT Fast5 API's Fast5Read
+    objects cannot be serialized through communication pipes with CPU workers.
+    This data is enough for worker to identify and load its input
+    data from fast5 file while avoiding costly transmision of raw data itself.
+    """
+
+    read_id: str
+    filename: str
+    caller: BasecallerInterface
```

Comment:
This is design flaw. It works fine with BasecallerMock, but can harm the performance once we have large class objects to be serailized. I'll rework this.